### ISSUE TITLE
201-vmss-linux-nat is broken

### ISSUE THREAD

**Issue by lastcoolnameleft** [Azure/azure-quickstart-templates#3011]
[201-vmss-linux-nat](https://github.com/Azure/azure-quickstart-templates/tree/master/201-vmss-linux-nat)

### Issue Details
When I try to create an instance based off the image, I get:

ERROR TYPE
The template deployment 'Microsoft.Template' is not valid according to the validation procedure. The tracking id is 'd3ce8f5a-e0a3-4513-afea-3bcc0d1a37dd'. See inner errors for details. Please see https://aka.ms/arm-deploy for usage details. (Code: InvalidTemplateDeployment)

Unfortunately, there appears to be another bug that prevents me from copy/pasting the error details, so this will be abbreviated:

Preflight validation failed.  Please refer to the details for the specific errors 
- asdfasdfasdf-mvssbsa is not a valid storage account name.  The storage account name must be between 3 and 24 characters in length and use numbers and lower-case letters only.  (CODE: AccountNameInvalid)

In short, creating the storage account breaks because it's based off the VMSS name, and the VMSS name allows "-", but the storage account does not.

Maybe other people will be smarter than me, but this caught me off guard as I spent some time figuring out the issue.

### Repro steps (*if necessary, delete otherwise*)
1. Go to https://github.com/Azure/azure-quickstart-templates/tree/master/201-vmss-linux-nat
2. Click "Deploy to Azure"
3. Choose a VMSS name with a "-".  e.g. "foo-bar"
4. Get Storage Account Name error.
5.
